app/server/src/main/scala/org/bitcoins/server/BitcoindRpcBackendUtil.scala

```diff
@@ -40,9 +44,10 @@ object BitcoindRpcBackendUtil extends Logging {
             _ <- lastConfirmedOpt match {
               case None =>
                 for {
-                  header <- bitcoind.getBestBlockHeader()
-                  _ <- wallet.stateDescriptorDAO.updateSyncHeight(header.hashBE,
-                                                                  header.height)
+                  _ <- doSync(walletHeight = 0,
```

Comment:
This changes an optimization in the wallet where if we had zero txs in the wallet, we would just update our wallet state to bitcoinds height. 

This needed to change to make sure we emit a `blockprocessed` event for every block

Reply:
why do we need to do this? could we just emit one for the chain tip?

Reply:
I took your suggestion, I think this is how to implement it? I basically just called `doSync()` on `bitcoind.height - 1`.